### PR TITLE
Connect API

### PR DESCRIPTION
**Pull request by limavfabio**
This PR links the bookstore state with the API following the Ducks proposal of organizing Redux store

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -1,47 +1,82 @@
-import { v4 as uuidv4 } from 'uuid';
+// import { v4 as uuidv4 } from 'uuid';
```
Comment (BertrandConxy) [inline src/redux/books/books.js]: - [x] Unnecessary commented-out codes should be removed from the codebase to make it look clean and smart. This is among the coding best practices. So please remove this line of code 😃 👍

File: src/redux/books/books.js [hunk 2]
```diff
@@ -1,47 +1,82 @@
...
+export const getBooks = () => async (dispatch) => {
+  const booksFetch = await fetch(
+    'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/z4gzcApO94uwV3qBSQiu/books',
+    {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+      },
+    }
+  ).then((data) => data.json());
```
Comment (BertrandConxy) [inline src/redux/books/books.js:30]: - [ ] Great job so far 👏. From the project requirements, you are expected the modern way to manage project data using different functions offered by the `redux-toolkit` like [CreateAsyncThunk](https://redux-toolkit.js.org/api/createAsyncThunk) and CreateSlice. All information about getting started and some examples can be found on the [redux-toolkit](https://redux-toolkit.js.org/introduction/getting-started) official documentation. What I would suggest is **to refactor your codes to use the method stated in the project requirements** 😃 👍

File: src/redux/books/books.js [hunk 3]
```diff
@@ -1,47 +1,82 @@
...
+  ).then(
+    dispatch({
+      type: ADD,
+      book,
+    })
+  );
+};
+
+export const removeBook = (book) => async (dispatch) => {
+  await fetch(
```
Comment (BertrandConxy) [inline src/redux/books/books.js:66]: - [ ] The same goes here and on all your action creators. Please try to use the method mentioned in the project requirements which is using CreateAsyncThunk and CreateSlice functions from the redux-toolkit library. I have added important links in the above review so, you can use them to implement it in all your action creators 😉

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @limavfabio, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 


## Highlights 💪 

- No linter errors
- correct flow is used
- good PR


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by EricMbouwe — APPROVED** [review]
Hi @limavfabio, 

Your project is complete! ⚡🔥  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights 🎉🔥
- Requested changes ✔

## Optional suggestions ☂

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**